src/components/calcite-input/calcite-input.tsx

```diff
@@ -394,6 +397,20 @@ export class CalciteInput {
     }
   };
 
+  private inputNumberInputHandler = (nativeEvent: InputEvent): void => {
+    const value = (nativeEvent.target as HTMLInputElement).value;
+    const delocalizedValue = delocalizeNumberString(value, this.locale);
+    if (nativeEvent.inputType === "insertFromPaste") {
+      if (!isValidNumber(delocalizedValue)) {
+        nativeEvent.preventDefault();
+      }
+      this.setValue(parseNumberString(delocalizedValue), nativeEvent);
+      this.childNumberEl.value = this.localizedValue;
```

Comment:
Doesn't `childNumberEl` display `localizedValue` without setting it?

Reply:
It's kinda complicated.  `localizedValue` is essentially a computed value that almost always is reflected verbatim in `childNumberEl.value` but because the input needs to allow typing decimals which results in a momentary invalid number (because numbers can't end with a period) the typing in the input is less restricted than it might otherwise need to be, and because of this the render method isn't always called when a user types into the input.

Reply:
Thanks for the info. Not sure if will help here, but you can force a render if necessary (w/ `forceUpdate`).

Reply:
`forceUpdate` doesn't cut it, tried that.  The only thing that actually works is directly setting the value like this.